external_deps/build-toolchain.sh

```diff
@@ -0,0 +1,209 @@
+#! /usr/bin/env bash
```

Comment:
Could this be Python? Compared to build.sh it seems like there are less simple shell commands and more management of temp directories etc. so maybe it would benefit.

Reply:
Yeah, maybe, I also thought about the same after having written this.